http_parser.c

```diff
@@ -472,6 +483,85 @@ static struct {
 
 int http_message_needs_eof(const http_parser *parser);
 
+const char* find_crlf(const char* p, const char* data, size_t len);
+
+#if defined(_MSC_VER) // SSE2 is baseline
+#include <intrin.h>
+#define __builtin_ctz _tzcnt_u32
+#elif defined(__SSE2__) && defined(__GNUC__)
+#include <emmintrin.h>
+#endif
+
+#if defined(_MSC_VER) || (defined(__SSE2__) && defined(__GNUC__))
```

Comment:
Perhaps it might be a good idea to make this an overridable define so that the plain C versions of functions can be tested more easily.

Reply:
Using `-mno-sse2` (regardless of `-march` or `-mtune`) would achieve this, perhaps less explicitly than you're aiming for though.

Reply:
I agree that overridable define should be introduced here.

Reply:
done, i hope this is what you guys are looking for :P